app/src/main/java/com/bogdan/codeforceswatcher/features/problems/ProblemsAdapter.kt

```diff
@@ -65,11 +69,49 @@ class ProblemsAdapter(
         }
     }
 
-    fun setItems(problemsList: List<Problem>) {
+    fun setItems(problemsList: List<Problem>, constraint: String) {
         items = problemsList
+        showingItems = buildFilteredList(constraint)
         notifyDataSetChanged()
     }
 
+    private val problemFilter: Filter = object : Filter() {
+        override fun performFiltering(constraint: CharSequence?): FilterResults {
+            val filteredList = mutableListOf<Problem>()
+            if (constraint.isNullOrEmpty()) {
+                filteredList.addAll(items)
+            } else {
+                filteredList.addAll(buildFilteredList(constraint.toString()))
+            }
+            val results = FilterResults()
+            results.values = filteredList
+            return results
+        }
+
+        override fun publishResults(constraint: CharSequence, results: FilterResults) {
+            showingItems.clear()
+            showingItems.addAll(results.values as List<Problem>)
+            notifyDataSetChanged()
+        }
+    }
+
+    private fun buildFilteredList(constraint: String): MutableList<Problem> {
+        val lowerCaseConstraint = constraint.toLowerCase(Locale.getDefault())
```

Comment:
Not sure locale is needed here and in this method overall. Why can't we just `lowercase` without it?

Reply:
It is needed - else -> warning